			     BASH PATCH REPORT
			     =================

Bash-Release:	5.1
Patch-ID:	bash51-002

Bug-Reported-by:	oguzismailuysal@gmail.com
Bug-Reference-ID:	<CAH7i3LoHFUa4aSF5-AD2r80HG-p-YzD_9ZxomarZkhP8NMq63g@mail.gmail.com>
Bug-Reference-URL:	https://lists.gnu.org/archive/html/bug-bash/2020-12/msg00037.html

Bug-Description:

If there are no jobs, and the `-n' and `-p' options are both supplied to
`wait', bash can assign a value to the variable name specified with `-p'
instead of leaving it unset.

Patch (apply with `patch -p0'):

--- a/builtins/wait.def
+++ b/builtins/wait.def
@@ -213,11 +213,11 @@ wait_builtin (list)
 	}
 
       status = wait_for_any_job (wflags, &pstat);
-      if (status < 0)
-	status = 127;
-
       if (vname && status >= 0)
 	bind_var_to_int (vname, pstat.pid);
+
+      if (status < 0)
+	status = 127;
       if (list)
 	unset_waitlist ();
       WAIT_RETURN (status);
--- a/patchlevel.h
+++ b/patchlevel.h
@@ -25,6 +25,6 @@
    regexp `^#define[ 	]*PATCHLEVEL', since that's what support/mkversion.sh
    looks for to find the patch level (for the sccs version string). */
 
-#define PATCHLEVEL 1
+#define PATCHLEVEL 2
 
 #endif /* _PATCHLEVEL_H_ */
